### DISCUSSION TITLE
Hardware USB keys

### DISCUSSION THREAD

**Discussion by arctus** [ebourg/jsign#186] [General]
Just a question before I go out and buy, as I understand the project supports only Yubikey dongle and does not support the SafeNet dongle? If that is the case, may you please refer me to any guide on how to stop Yubikey from asking for a PIN (to be included in a CI environment)?

**Comment by ebourg** [comment]
Jsign works with the SafeNet eToken using the `PKCS11` store type, it simply requires a [SunPKCS11 configuration file](https://docs.oracle.com/javase/8/docs/technotes/guides/security/p11guide.html#Config).

I don't think you can disable the PIN, but that's not really an issue, when invoking Jsign you can pass it using the `--storepass` parameter.

**Reply by arctus** [reply]
Thank you, appreciate the response and your help.

**Reply by ebourg** [reply]
I've added a new `ETOKEN` store type to better support the SafeNet eToken. The syntax looks like this:

    jsign --storetype ETOKEN --storepass <PIN> --certfile full-chain.pem application.exe

If you get an eToken please let me know how it works for you.

**Reply by arctus** [reply]
thank you!
